pandas/core/arrays/categorical.py

```diff
@@ -1158,6 +1164,17 @@ def remove_categories(self, removals, inplace=False):
         remove_unused_categories : Remove categories which are not used.
         set_categories : Set the categories to the specified ones.
         """
+        if inplace is not no_default:
+            warn(
+                "The `inplace` parameter in pandas.Categorical."
+                "remove_categories is deprecated and "
+                "will be removed in a future version.",
```

Comment:
usually we would say "do X instead", but since the idea is to never change the dtype inplace, might be worth adding a short sentence like "Removing unused categories will always return a new Categorical object"

Reply:
Ok, I updated the warning message.